sda-commons-web-autoconfigure/src/test/java/org/sdase/commons/spring/boot/web/security/test/ContextUtils.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2022- SDA SE Open Industry Solutions (https://www.sda.se)
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+package org.sdase.commons.spring.boot.web.security.test;
+
+import org.springframework.boot.test.context.SpringBootContextLoader;
+import org.springframework.boot.test.context.SpringBootTestContextBootstrapper;
+import org.springframework.boot.test.context.assertj.AssertableApplicationContext;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate;
+import org.springframework.test.context.support.DefaultBootstrapContext;
+
+public class ContextUtils {
+  private ContextUtils() {
+    // avoid instantiation
+  }
+
+  /**
+   * Creates a spring test context for testing application startup failures. Using `SpringBootTest`
+   * annotation will prevent the tests from starting. Tests that need dependencies from the web
+   * environment must add `@SpringBootTest(webEnvironment = ...)` to the Spring boot application
+   * class because that is how it is internally checked.
+   *
+   * @param clazz The application class annotated with `SpringBootApplication`
+   * @return An assertable application context.
+   */
```

Comment:
How are you supposed to test the startup failure? Example would help.

Reply:
added